graph/multi/undirected.go

```diff
@@ -0,0 +1,250 @@
+// Copyright ©2014 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multi
+
+import (
+	"fmt"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/internal/uid"
+)
+
+var (
+	ug *UndirectedGraph
+
+	_ graph.Graph                = ug
+	_ graph.Undirected           = ug
+	_ graph.Multigraph           = ug
+	_ graph.UndirectedMultigraph = ug
+)
+
+// UndirectedGraph implements a generalized undirected graph.
+type UndirectedGraph struct {
+	nodes map[int64]graph.Node
+	lines map[int64]map[int64]map[int64]graph.Line
+
+	nodeIDs uid.Set
+	lineIDs uid.Set
+}
+
+// NewUndirectedGraph returns an UndirectedGraph with the specified self and absent
```

Comment:
No weights here.

Reply:
I'll wait for the rest of the review before I fix this.

Reply:
Done